pmlc/testing/gtest_main.cc

```diff
@@ -17,6 +21,38 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
+  if (FLAGS_skip_test_file.size()) {
+    // If there's a skip test file, we're going to take each pattern it contains,
+    // and add those patterns to gtest's filter's negative pattern list.
+
+    // To start, see whether there's a '-' in the existing filter.  If
+    // there isn't, we'll need to add one (to indicate that these are
+    // negative tests); if there is, we should use ':'.  Regardless,
+    // each subsequent added pattern is separated by ':'.
+    char sep = ':';
+    if (testing::FLAGS_gtest_filter.find('-') == std::string::npos) {
+      sep = '-';
+    }
+
+    // Process the skip pattern file, building the flags pattern.
+    std::ifstream fs{FLAGS_skip_test_file};
+    if (!fs) {
+      std::cerr << "Unable to open skip test file '" << FLAGS_skip_test_file << "'\n";
+      return EXIT_FAILURE;
+    }
+
+    std::ostringstream ss{testing::FLAGS_gtest_filter};
+
+    for (std::string line; std::getline(fs >> std::ws, line);) {
+      // Erase trailing whitespace
+      line.erase(std::find_if_not(line.rbegin(), line.rend(), [](int c) -> bool { return std::isspace(c); }).base(), line.end());
+      ss << sep << line;
+      sep = ':';
+    }
+
+    testing::FLAGS_gtest_filter = ss.str();
+  }
+
```

Comment:
How does this interact with existing filters? Let's say I want to try out a 'skipped' test, is there a way to override the filter so that the one from the bazel command line takes precedence?

Reply:
Omit the filter file and specify the skip filter explicitly (which is easy, because the test output contains the full setting after the filter file's been applied).  It's hard to do this any other way, since the filters are patterns.

FWIW, I also thought about allowing comments in the file format, so that we could describe why things were being omitted -- YAGNI for now, though.